packages/@aws-cdk/aws-config/lib/rule.ts

```diff
@@ -352,11 +352,14 @@ export class CustomRule extends RuleNew {
         messageType: 'ScheduledNotification',
       });
     }
-
-    props.lambdaFunction.addPermission('Permission', {
-      principal: new iam.ServicePrincipal('config.amazonaws.com'),
-      sourceAccount: this.env.account,
-    });
+    const customRuleFnName: string = props.lambdaFunction.node.id;
+    const customRulePermissionId: string = `customRulePermission-${customRuleFnName}`;
+    if (!props.lambdaFunction.permissionsNode.tryFindChild(customRulePermissionId)) {
+      props.lambdaFunction.addPermission(customRulePermissionId, {
+        principal: new iam.ServicePrincipal('config.amazonaws.com'),
+        sourceAccount: this.env.account,
```

Comment:
I think this wouldn't work if we had multiple config rules in different accounts. This would only add the permission for the first account. I think we somehow need to add the account into the id.

Reply:
And actually I wonder if it would be better to handle this in the `addPermission` method since this problem is probably not unique to config. We could probably do something similar to what we do for `grantInvoke`

https://github.com/aws/aws-cdk/blob/8a733bad022fb2ac0551e32fd26a4847c0b36ff2/packages/%40aws-cdk/aws-lambda/lib/function-base.ts#L433-L414

Reply:
Thanks for the quick review!
It seems more natural to use a random hash rather than forcing it to get an accountid, and it seems to be more natural to use a random hash, and the ids don't conflict. I will give it a try.

Reply:
Addendum:.
I have created a hash from FunctionName, AccountID, and Region to make the suffix unique.
Therefore, the omitted parts in the test code have been modified to fix the result.

Reply:
@watany-dev what do you think about my suggestion for doing this in the `addPermission` method instead? That would be a more global fix for this issue.

Reply:
yeah I see your point. Ok i'm good with your solution, just a couple of comments on the integ tests.